compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/PropagateScopeDependencies.ts

```diff
@@ -744,51 +747,212 @@ class PropagationVisitor extends ReactiveFunctionVisitor<Context> {
+     * ```
+     */
+    if (
+      sequence.instructions.length === 1 &&
+      sequence.instructions[0].value.kind === 'LoadLocal' &&
+      sequence.instructions[0].lvalue !== null &&
+      sequence.instructions[0].value.place.identifier.name !== null &&
+      !context.isUsedOutsideDeclaringScope(sequence.instructions[0].lvalue) &&
+      sequence.value.kind === 'SequenceExpression' &&
+      sequence.value.instructions.length === 1 &&
+      sequence.value.instructions[0].value.kind === 'PropertyLoad' &&
+      sequence.value.instructions[0].value.object.identifier.id ===
+        sequence.instructions[0].lvalue.identifier.id &&
+      sequence.value.instructions[0].lvalue !== null &&
+      sequence.value.value.kind === 'LoadLocal' &&
+      sequence.value.value.place.identifier.id ===
+        sequence.value.instructions[0].lvalue.identifier.id
+    ) {
+      context.declareTemporary(
+        sequence.instructions[0].lvalue,
+        sequence.instructions[0].value.place,
+      );
+      const propertyLoad = sequence.value.instructions[0].value;
+      return {
+        lvalue,
+        object: propertyLoad.object,
+        property: propertyLoad.property,
+        optional: optionalValue.optional,
+      };
+    }
+    /**
+     * Composed case: `<base-case> "." or "?." <property>`
+     *
+     * This case is convoluted, note how `t0` appears as an lvalue *twice*
+     * and then is an operand of an intermediate LoadLocal and then the
+     * object of the final PropertyLoad:
+     *
+     * ```
+     * <lvalue> = OptionalExpression optional=false (`optionalValue` is here)
+     *  Sequence (`sequence` is here)
+     *      t0 = Sequence
+     *        t0 =
+     *           <nested>
+     *        LoadLocal t0
+     *      Sequence
+     *        t1 = PropertyLoad t0. <property>
+     *        LoadLocal t1
+     * ```
+     */
+    if (
+      sequence.instructions.length === 1 &&
+      sequence.instructions[0].value.kind === 'SequenceExpression' &&
+      sequence.instructions[0].value.instructions.length === 1 &&
+      sequence.instructions[0].value.instructions[0].lvalue !== null &&
+      sequence.instructions[0].value.instructions[0].value.kind ===
+        'OptionalExpression' &&
+      sequence.instructions[0].value.value.kind === 'LoadLocal' &&
+      sequence.instructions[0].value.value.place.identifier.id ===
+        sequence.instructions[0].value.instructions[0].lvalue.identifier.id &&
+      sequence.value.kind === 'SequenceExpression' &&
+      sequence.value.instructions.length === 1 &&
+      sequence.value.instructions[0].lvalue !== null &&
+      sequence.value.instructions[0].value.kind === 'PropertyLoad' &&
+      sequence.value.instructions[0].value.object.identifier.id ===
+        sequence.instructions[0].value.value.place.identifier.id &&
+      sequence.value.value.kind === 'LoadLocal' &&
+      sequence.value.value.place.identifier.id ===
+        sequence.value.instructions[0].lvalue.identifier.id
```

Comment:
what

Reply:
the

Reply:
This is bonkers but also temporary, we won’t need this once PropagateScopeDeps is HIR-based. So seems fine for now - worst case it doesn’t match and we fall back to the previous behavior.